GraphicsProgram/Form1.cs

```diff
@@ -43,63 +52,40 @@ private void exitToolStripMenuItem_Click(object sender, EventArgs e)
             Application.Exit();
         }
 
-
-        
-
         private void runbutton_Click(object sender, EventArgs e)
         {
             //Update X & Y coardinte text fields on run command
-            int updateX = penX;
-            int updateY = penY;
-
-            string stringX = updateX.ToString();
-            string stringY = updateY.ToString();
-
-            textBox3.Text = stringX;
-            textBox2.Text = stringY;
-
+            var penPosition = new PenPosition();
+            penPosition.X = penX;
+            penPosition.Y = penY;
 
+            textBox3.Text = penPosition.GetXAsString();
+            textBox2.Text = penPosition.GetYAsString();
+            
             //String array to split multi line text input
             string[] textBoxLines = userinput.Lines;
 
             string[] commands = new string[] { "repeat", "loop", "if" };
-
-
-
+                       
             //Input parsing, split multilines to single lines then split the single lines into an array
             foreach (string line in textBoxLines)
             {
                 var splitString = line.Split();
 
                 if (line.Contains("circle"))
                 {
-                    var number = splitString[1];
-                    if (double.TryParse(number, out circleRadius))
-                    {
-                        Circle circle = new Circle(circleRadius);
-                        double diameter = circle.getDiameter;
-                        float diameterF = Convert.ToSingle(diameter);
-                        g.DrawEllipse(myPen, penX, penY, diameterF, diameterF);
-                    }
-                    else
-                    {
-                        MessageBox.Show("Circle Parse Error");
-                    }
-
+                    _userOperationStrategies.Single(x => x.AppliesTo(OperationType.If, ShapeType.Circle)).DoDrawing(myPen, penPosition, g, line);
```

Comment:
It's around here that you should look at adding the different operations (`if`, `loop`, `repeat` etc etc)